core/src/main/java/org/mule/TransformationService.java

```diff
@@ -98,32 +100,14 @@ public MuleMessage applyTransformers(final MuleMessage message, final MuleEvent
             result = applyAllTransformers(message, event, transformers);
         }
 
-        if (null != outputType && !result.getPayload().getClass().isAssignableFrom(outputType))
+        if (outputType != null)
```

Comment:
Should not check if outputType is compatible with result.getPayload.getDatatype instead of checking for non null?